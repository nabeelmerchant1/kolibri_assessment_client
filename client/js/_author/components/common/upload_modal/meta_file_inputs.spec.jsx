import React            from 'react';
import { shallow }      from 'enzyme';
import MetaFileInputs   from './meta_file_inputs';

describe('MetaFileInputs component', () => {
  let props;
  let result;
  let updateFunction;

  beforeEach(() => {
    updateFunction = false;
    props = {
      metadataFileTypes: ['Spectastic'],
      labelName: () => {},
      updateMetadata: () => { updateFunction = true; },
    };
    result = shallow(<MetaFileInputs {...props} />);
  });

  it('matches the snapshot taken', () => {
    expect(result).toMatchSnapshot();
  });

  it('calls updateMetadata function', () => {
    expect(updateFunction).toBeFalsy();
    result.find('input').simulate('change', { target: { files: 'SPEC' } });
    expect(updateFunction).toBeTruthy();
  });
});
