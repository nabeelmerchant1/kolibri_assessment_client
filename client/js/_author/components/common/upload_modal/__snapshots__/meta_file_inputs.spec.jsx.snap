// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaFileInputs component matches the snapshot taken 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
        className="au-u-mb-md au-u-mt-md"
    >
        <label
            className="au-c-input--file au-o-flex-center"
            htmlFor="metadata_input_Spectastic"
        >
            <span
                className="au-c-wysiwyg-media__label"
            />
            <div
                className="au-c-wysiwyg-media__source-text"
                tabIndex="0"
            />
            <input
                id="metadata_input_Spectastic"
                onChange={[Function]}
                type="file"
            />
            <div
                className="au-c-wysiwyg-media__upload"
            >
                <i
                    className="material-icons"
                >
                    find_in_page
                </i>
            </div>
        </label>
    </div>
</div>,
  "nodes": Array [
    <div>
      <div
            className="au-u-mb-md au-u-mt-md"
      >
            <label
                  className="au-c-input--file au-o-flex-center"
                  htmlFor="metadata_input_Spectastic"
            >
                  <span
                        className="au-c-wysiwyg-media__label"
                  />
                  <div
                        className="au-c-wysiwyg-media__source-text"
                        tabIndex="0"
                  />
                  <input
                        id="metadata_input_Spectastic"
                        onChange={[Function]}
                        type="file"
                  />
                  <div
                        className="au-c-wysiwyg-media__upload"
                  >
                        <i
                              className="material-icons"
                        >
                              find_in_page
                        </i>
                  </div>
            </label>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <metaDataFileInputs
        labelName={[Function]}
        metadataFileTypes={
                Array [
                        "Spectastic",
                      ]
        }
        updateMetadata={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "labelName": [Function],
          "metadataFileTypes": Array [
            "Spectastic",
          ],
          "updateMetadata": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
                    className="au-u-mb-md au-u-mt-md"
          >
                    <label
                              className="au-c-input--file au-o-flex-center"
                              htmlFor="metadata_input_Spectastic"
                    >
                              <span
                                        className="au-c-wysiwyg-media__label"
                              />
                              <div
                                        className="au-c-wysiwyg-media__source-text"
                                        tabIndex="0"
                              />
                              <input
                                        id="metadata_input_Spectastic"
                                        onChange={[Function]}
                                        type="file"
                              />
                              <div
                                        className="au-c-wysiwyg-media__upload"
                              >
                                        <i
                                                  className="material-icons"
                                        >
                                                  find_in_page
                                        </i>
                              </div>
                    </label>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
                    className="au-u-mb-md au-u-mt-md"
          >
                    <label
                              className="au-c-input--file au-o-flex-center"
                              htmlFor="metadata_input_Spectastic"
                    >
                              <span
                                        className="au-c-wysiwyg-media__label"
                              />
                              <div
                                        className="au-c-wysiwyg-media__source-text"
                                        tabIndex="0"
                              />
                              <input
                                        id="metadata_input_Spectastic"
                                        onChange={[Function]}
                                        type="file"
                              />
                              <div
                                        className="au-c-wysiwyg-media__upload"
                              >
                                        <i
                                                  className="material-icons"
                                        >
                                                  find_in_page
                                        </i>
                              </div>
                    </label>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <metaDataFileInputs
    labelName={[Function]}
    metadataFileTypes={
        Array [
            "Spectastic",
          ]
    }
    updateMetadata={[Function]}
/>,
}
`;
